app/src/main/java/nl/tudelft/pixelperfect/pixelperfect/minigame/FireOutbreakActivity.java

```diff
@@ -0,0 +1,172 @@
+package nl.tudelft.pixelperfect.pixelperfect.minigame;
+
+import android.os.Bundle;
+import android.preference.CheckBoxPreference;
+import android.support.v7.app.AppCompatActivity;
+import android.view.View;
+import android.widget.Button;
+import android.widget.CheckBox;
+import android.widget.ProgressBar;
+
+import nl.tudelft.pixelperfect.client.GameClient;
+import nl.tudelft.pixelperfect.client.message.EventCompletedMessage;
+import nl.tudelft.pixelperfect.event.type.EventTypes;
+import nl.tudelft.pixelperfect.pixelperfect.R;
+
+/**
+ * Class responsible for the handling of the fire outbreak event.  A choice must
+ * be made between multiple solutions, and then selected where to
+ * deploy them to combat the fire.
+ *
+ * @author David Alderliesten
+ *
+ */
+public class FireOutbreakActivity extends AppCompatActivity {
+    private GameClient game;
+
+    private int fireLocation;
+
+    private CheckBox useFireExtinguisher;
+    private CheckBox useAutomator;
+    private CheckBox useWaterBucket;
+    private CheckBox useFireBlanket;
+    private CheckBox useHammer;
+
+    private Button deployToAirSupport;
+    private Button deployToEngine;
+    private Button deployToWingPlasma;
+
+    /**
+     * On creation of the fire activity the content views will be initialized
+     * and relevant information retrieved.
+     *
+     * @param savedInstanceState the instance to create.
+     */
+    @Override
+    protected void onCreate(Bundle savedInstanceState) {
+        super.onCreate(savedInstanceState);
+        setContentView(R.layout.activity_fire_outbreak);
+
+        useFireExtinguisher = (CheckBox) findViewById(R.id.extinguishBox);
+        useAutomator = (CheckBox) findViewById(R.id.automatedKillerBox);
+        useWaterBucket = (CheckBox) findViewById(R.id.bucketBox);
+        useFireBlanket = (CheckBox) findViewById(R.id.blanketBox);
+        useHammer = (CheckBox) findViewById(R.id.hammerBox);
+
+        deployToAirSupport = (Button) findViewById(R.id.airSupportButton);
+        deployToEngine = (Button) findViewById(R.id.engineButton);
+        deployToWingPlasma = (Button) findViewById(R.id.wingPlasmaButton);
+
+        game = GameClient.getInstance();
+    }
+
+    /**
+     * Class that validates the user action and returns the result to the server.
+     *
+     * @param passedState
+     *      The given state of validation.
+     */
+    private void validateAction(int passedState) {
+        game.sendMessage(new EventCompletedMessage(EventTypes.FIRE_OUTBREAK.ordinal()));
+        finishActivity(0);
```

Comment:
According to the official Android docs, an intent is counted as an activity.  When you call finishActivity 0, you're indicating no errors occurred (hence the 0 code, as in C), and the finishACtivity returns you to where you were.  This is needed because Fire can be done by any role, and not just one role as with all others mini-games.